test_phage_reference/nbconverted/1_align_quantify_samples.py

```diff
@@ -0,0 +1,211 @@
+# 2. Computing the maximum mappable prefix (MMP) of the query beginning with this k-mer
+# 3. Determining the next informative position (NIP) by performing a longest common prefix (LCP) query on two specifically chosen suffixes in the SA
+
+# In[9]:
+
+
+# Get PAO1 index
+get_ipython().system(' salmon index -t $paths.PAO1_REF -i $paths.PAO1_INDEX')
+
+
+# In[10]:
+
+
+# Get phage index
+get_ipython().system(' salmon index -t $paths.PHAGE_REF -i $paths.PHAGE_INDEX')
+
+
+# ### Quantify gene expression
+# Now that we have our index built and all of our data downloaded, we’re ready to quantify our samples
+# 
+# **Input:**
+# * Index of reference transcriptome
+# * FASTQ of experimental samples
+# 
+# **Output:**
+# 
+# After the salmon commands finish running, you should have a directory named quants, which will have a sub-directory for each sample. These sub-directories contain the quantification results of salmon, as well as a lot of other information salmon records about the sample and the run. 
+# 
+# The main output file (called `quant.sf`). Inside the quantification file for sample DRR016125 in quants/DRR016125/quant.sf, you’ll see a TSV format file listing the name (`Name`) of each transcript, its length (`Length`), effective length (`EffectiveLength`), and its abundance in terms of Transcripts Per Million (`TPM`) and estimated number of reads (`NumReads`) originating from this transcript.
+# 
+# **For each sample we have read counts per gene (where the genes are based on the reference gene file provided above).** 
+
+# #### Get quants using PAO1 reference
+
+# In[ ]:
+
+
+if not os.path.exists(paths.PAO1_QUANT):
+    os.makedirs(paths.PAO1_QUANT)
+
+
+# In[13]:
+
+
+get_ipython().run_cell_magic('bash', '-s $paths.PAO1_QUANT $paths.FASTQ_DIR $paths.PAO1_INDEX', '\nfor FILE_PATH in $2/*;\ndo\n\n# get file name\nsample_name=`basename ${FILE_PATH}`\n\n# remove extension from file name\nsample_name="${sample_name%_*}"\n\n# get base path\nbase_name=${FILE_PATH%/*}\n\necho "Processing sample ${sample_name}"\n\nsalmon quant -i $3 -l A \\\n            -1 ${base_name}/${sample_name}_1.fastq \\\n            -2 ${base_name}/${sample_name}_2.fastq \\\n            -p 8 --validateMappings -o $1/${sample_name}_quant\ndone')
+
+
+# #### Get quants using phage reference
+
+# In[ ]:
+
+
+if not os.path.exists(paths.PHAGE_QUANT):
+    os.makedirs(paths.PHAGE_QUANT)
+
+
+# In[12]:
+
+
+get_ipython().run_cell_magic('bash', '-s $paths.PHAGE_QUANT $paths.FASTQ_DIR $paths.PHAGE_INDEX', '\nfor FILE_PATH in $2/*;\ndo\n\n# get file name\nsample_name=`basename ${FILE_PATH}`\n\n# remove extension from file name\nsample_name="${sample_name%_*}"\n\n# get base path\nbase_name=${FILE_PATH%/*}\n\necho "Processing sample ${sample_name}"\n\nsalmon quant -i $3 -l A \\\n            -1 ${base_name}/${sample_name}_1.fastq \\\n            -2 ${base_name}/${sample_name}_2.fastq \\\n            -p 8 --validateMappings -o $1/${sample_name}_quant\ndone')
```

Comment:
Now this is just an opinion thing, but is there a reason not to just make these steps a bash script? It seems unwieldy to have all of this in a python command, especially with the '\n' stuff.

Reply:
Your intuition is right on! This should be a bash script. These are meant to be some quick test scripts to see what's going on with the data and use this information to update the working pipeline that is on Discovery. Since I'm more familiar with python this is what I defaulted to when I was quickly trying to put together some tests. But these will eventually be run as scripts on a cluster